### ISSUE TITLE
Auto-expansion of `from` path for ES6 imports

### ISSUE THREAD

**Issue by brianmhunt** [Microsoft/vscode#53894]
- VSCode Version: 1.25.0
- OS Version: All

Steps to Reproduce:

1. Create a file with `import {isObject} from 'lodash-es'`.
2. Save some other, seemingly unrelated files.
3. The `from` portion of the `import` is expanded in every file it occurs in to a relative import path, such as:

```javascript
import {isObject} from '../../../../Library/Caches/typescript/2.9/node_modules/@types/lodash-es'
```

Obviously this is just a minor nuisance when you've got a couple files, but when you hit 4-5 thousand files being updated for no apparent reason on a semi-regular basis, this is a major problem.

Does this issue occur when all extensions are disabled?: Yes

**Comment by luchsamapparat** [comment]
I'm experiencing similar issues in all of my Angular projects (all created with Angular CLI, if that's any help).

Another weird thing, I cannot quite reproduce: Sometimes, after opening a new folder in VS Code, it just starts opening TS files and modifying their imports without them being in any way related to the file I'm currently working on. At one point, ~80 files were opened and modified.

**Comment by mjbvz** [comment]
Please try upgrading your workspace to use `typescript@next` by following [these instructions](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions). Do you still see the issue?

**Comment by Ky6uk** [comment]
@mjbvz it isn't reproducing on the TS `3.0.0-dev.20180707`.

**Comment by brianmhunt** [comment]
I have upgrade to typescript 3.0.0-dev.20180707; I'll report back if I encounter the issue (which was quite frequent).

If you have not heard from me by next week this time, it'll be safe to presume the issue has not exhibited with this version of typescript for us.

Thanks

**Comment by brianmhunt** [comment]
Unfortunately the issue continues to exhibit with typescript@next (after having installed it with `yarn add -D typescript@next` and restarting vscode) by performing the following steps:

1. Have files that import from `lodash-es`
1. Have an import named `abc/abc.js` (the 'react' style of module import)
2. Have another file that has `import something from 'abc'`

If you rename `abc.js` in the file explorer with right-click, then all the `lodash-es` imports get rewritten to relative paths.

**Comment by clayne11** [comment]
My whole team is seeing this issue as well.

**Comment by mjbvz** [comment]
@brianmhunt Did you also run the `TypeScript: Select typescript version` command to change your workspace to use TS 3.0 as well?

VS Code insiders now also includes TS 3.0 by default. I can't repo this issue on 3.0 following your steps

**Comment by clayne11** [comment]
I tried the insiders version of VSCode with TS 3.0 and this was fixed for me.

**Comment by Curdflappers** [comment]
VS Code - Insiders with TS 3.0 also fixed the issue for me, including the auto-import hint.

**Comment by samtsai** [comment]
Selecting "Use Workspace Version" from the Typescript Status Bar fixed the issue for me that I'm experiencing in an Angular application. Though, previous versions of VS Code worked with `2.9.2`.

**Comment by tuhinkarmakar** [comment]
@samtsai I had to make the same changes to fix the issue. But the downside is, auto update for file/directory renaming/moving stopped working. TypeScript >= 2.9 is required for auto update to work :disappointed:

**Comment by brianmhunt** [comment]
I can confirm this still occurs, with a few debugging notes below.

Here's the output from some commands:

```
$ yarn list | grep typescript
├─ typescript@3.0.0-dev.20180707
$ cat .tsserver.js
{
   "typescript.tsdk": "./node_modules/typescript/lib"
}
```

There are no extensions of interest, and all the typescript settings are defaults.

Note that I could not find a `TypeScript: Select typescript version` option, nor a Typescript Status Bar (hence no `Use Workspace Version` option).

Please advise of any further information that might assist diagnosis & prescription.

**Comment by Curdflappers** [comment]
@brianmhunt what version of VS Code are you using? Have you tried to reproduce this with Code - Insiders?

**Comment by brianmhunt** [comment]
@Curdflappers VS Code 1.25.1.   I don't know what Code - Insiders is. :)

**Comment by Curdflappers** [comment]
Visual Studio Code - Insiders is the daily update version of VS Code, so it includes the most recent stable build, and you can update daily. For me, changing from regular VS Code to Insiders fixed this issue, though it was likely the typescript version. You can download Insiders here: https://code.visualstudio.com/insiders/

You should be able to copy and paste workspace settings, keyboard shortcuts, etc since they're JSON

**Comment by mjbvz** [comment]
I'm closing this issue as fixed in VS Code insiders when using TS 3.0. If you still see this issue when using typescript 3.0, please open a new issue and I'll take a look

**Comment by onmyway133** [comment]
I'm having this with VSC 1.25.1 on a pure Javascript React project

**Comment by tuhinkarmakar** [comment]
@mjbvz Just upgraded to VS Code 1.26. When using TypeScript 3.0.1, I'm not getting quick fixes for missing imports. If I switch back to TypeScript 2.9.2 (which Angular is using), I'm getting the quick fixes but the expansion issue remains.

**Comment by mjbvz** [comment]
@tuhinkarmakar Unrelated to the original issue. Likely https://github.com/Microsoft/TypeScript/issues/26333 Try upgrading your workspace to TypeScript@next to see if the issue is fixed

**Comment by tuhinkarmakar** [comment]
@mjbvz Yes, you are right. The problem I reported is unrelated to the original issue. Like you [said](https://github.com/Microsoft/vscode/issues/53894#issuecomment-406133367), TypeScript 3.0 did fix the original issue. But after I upgraded to version 3.0, I faced the issue mentioned above. I just wanted to leave a note here that upgrading to version 3.0 has its own problems.

**P.S.** Though I am not sure, I think issue #57017 and #57164 are related to the issue I am facing.
